mypy/checkexpr.py

```diff
@@ -1902,10 +1902,16 @@ def analyze_ordinary_member_access(self, e: MemberExpr,
         else:
             # This is a reference to a non-module attribute.
             original_type = self.accept(e.expr)
+            base = e.expr
+            module_members = None
+
+            if isinstance(base, RefExpr) and isinstance(base.node, MypyFile):
+                module_members = [key for key in base.node.names.keys() if not base.node.names.get(key).module_hidden]
```

Comment:
This can be expensive to calculate, in case the module has many attributes. Instead of passing the members, what about passing a `SymbolTable` object, and calculating the members only later if we generate an error message?